eth/client.go

```diff
@@ -938,6 +935,63 @@ func (c *client) ReplaceTransaction(tx *types.Transaction, method string, gasPri
 	return newSignedTx, err
 }
 
+func (c *client) WatchForJob(streamId string) (*lpTypes.Job, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), c.txTimeout)
+	defer cancel()
+	var job *lpTypes.Job
+	jobWatcher := func() error {
+		sink := make(chan *contracts.JobsManagerNewJob)
+		sub, err := c.JobsManagerSession.Contract.JobsManagerFilterer.WatchNewJob(nil, sink, []ethcommon.Address{c.Account().Address})
+		if err != nil {
+			glog.Error("Unable to start job watcher ", err)
+			return err
+		}
+		select {
+		case newJob := <-sink:
+			sub.Unsubscribe()
+			if newJob.StreamId == streamId {
+				// might be faster to reconstruct the job locally?
+				j, err := c.GetJob(newJob.JobId)
+				if err != nil {
+					glog.Error("Unable to fetch job after watching: ", err)
+					// maybe perform/retry the job lookup outside this loop
+					// but a retry may be unlikely to succeed, depending on the error
+					// Manually create the job for now.
+					// May have important fields missing!!!
```

Comment:
IMO if `WatchForJob` is expected to be able to be reused by anyone then we should always return a job with all the fields populated (retrieved using a RPC call) and if it is expected to be only used by a broadcaster then we should always return a job partially filled with broadcaster related fields (i.e. reconstructed from the fields in the event). I don't think it is a problem right now since `WatchForJob` is used exclusively in `CreateTranscodeJob` which is only relevant for a broadcaster, but I think it could be confusing for usage later on if the function ends up being used elsewhere and it sometimes returns a job with all the fields and sometimes returns a job with only some of the fields

Reply:
We could also refactor the `JobService` to use `WatchForJob`  rather than the old eventmonitor (which would also not expect have all fields populated) but agreed that there's potential to 'surprise' API users later on. We can add a note to the function warning future users about the potential discrepancy.

Reply:
Note added:

https://github.com/livepeer/go-livepeer/blob/35f32fd32e3a03d3a598c57fe7f71fed8a3c49e1/eth/client.go#L938-L941